src/lib/components/table/table-data-source.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { TableDataSource } from './table-data-source';
+import { RequestParameterInterface } from './request-parameter.interface';
+import { TerraPagerInterface } from '../pager/data/terra-pager.interface';
+import {
+    Observable,
+    of
+} from 'rxjs';
+import { TerraFilter } from './filter';
+
+class TestDataSource extends TableDataSource<{}>
+{
+    public request(requestParams:RequestParameterInterface):Observable<Array<{}>> | Observable<TerraPagerInterface<{}>>
+    {
+        return of(new Array({}));
+    }
+}
+
+interface FilterParams
+{
+    id:number;
+}
+
+describe('TableDataSource', () =>
+{
+    let dataSource:TestDataSource;
+    let filter:TerraFilter<FilterParams>;
+
+    beforeEach(() =>
+    {
+        dataSource = new TestDataSource();
+        filter = new TerraFilter<FilterParams>();
+    });
+
+    it('should create', () =>
+    {
+        expect(dataSource).toBeTruthy();
+    });
+
+    it('should create an observable for a request', () =>
+    {
+        expect(dataSource.request({})).toBeTruthy();
+    });
+
+    it('should have a filter', () =>
+    {
+        filter.filterParameter = {id: 1};
+        dataSource.filter = filter;
+
+        expect(dataSource.filter).toEqual(filter);
+    });
```

Comment:
Das ist nicht nötig. Tested quasi ob der Zuweisungs-Operator von Javascript richtig funktioniert 😄

Reply:
Ja und nein. Es wird auch getestet, dass in der Klasse nichts anderes mehr mit dem Filter gemacht wird. Der Test stellt auch sicher, dass bei zukünftigen Änderungen (getter, wo eine Kopie des Filters erzeugt wird) eben das gleiche Filterobjekt erwartet wird.